.github/workflows/docker.yml

```diff
@@ -1,55 +1,39 @@
-name: Build & Push
-# Build & Push builds the simapp docker image on every push to master and
-# and pushes the image to https://hub.docker.com/r/interchainio/simapp/tags
+name: Publish Docker image
 on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
   pull_request:
-  push:
-    branches:
-      - main
-    tags:
-      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
-      - "v[0-9]+.[0-9]+.[0-9]+-rc*" # Push events to matching v*, i.e. v1.0-rc1, v20.15.10-rc5
-
+  push: 
+  release:
+    types: 
+      - created
+      
 jobs:
-  build:
+  push_to_registry:
+    name: Push Multiplatform Gaia Docker image to GitHub Packages
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@master
-        with:
-          fetch-depth: 0
-
-      - name: Prepare
-        id: prep
-        run: |
-          DOCKER_IMAGE=tendermint/gaia
-          VERSION=noop
-          if [[ $GITHUB_REF == refs/tags/* ]]; then
-            VERSION=${GITHUB_REF#refs/tags/}
-          elif [[ $GITHUB_REF == refs/heads/* ]]; then
-            VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
-            if [ "${{ github.event.repository.default_branch }}" = "$VERSION" ]; then
-              VERSION=latest
-            fi
-          fi
-          TAGS="${DOCKER_IMAGE}:${VERSION}"
-          if [[ $VERSION =~ ^v[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
-            TAGS="$TAGS,${DOCKER_IMAGE}:${VERSION}"
-          fi
-          echo ::set-output name=version::${VERSION}
-          echo ::set-output name=tags::${TAGS}
-          echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+    
+      - name: GitHub Environment Variables Action
+        uses: FranzDiebold/github-env-vars-action@v2.1.0
 
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
+      - name: Check out the repo
+        uses: actions/checkout@v2
+        
+      - name: Set up QEMU
+        run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes --credential yes
+        
+      - name: Login to Packages Registry
+        uses: docker/login-action@v1 
         with:
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Publish to Docker Hub
+      - name: Push to GitHub Packages
         uses: docker/build-push-action@v2
         with:
-          push: ${{ github.event_name != 'pull_request' }}
-          tags: ${{ steps.prep.outputs.tags }}
+          context: .
+          platforms: linux/amd64,linux/arm64
+          registry: docker.pkg.github.com
+          repository: ${{ github.actor }}/gaia/gaia
+          tags: ${GITHUB_REPOSITORY}:${BRANCH_NAME},${GITHUB_REPOSITORY}:${BRANCH_NAME}
```

Comment:
does this work for releases?

Reply:
Not yet, actually it should have contained a comment/todo:

* Need to tag releases.

Reply:
'bout an hour later, I guess that it'll work for releases